applications/LinearSolversApplication/custom_solvers/eigen_dense_householder_qr_solver.h

```diff
@@ -23,7 +23,7 @@ class EigenDenseHouseholderQRSolver
 {
 public:
     using Scalar = TScalar;
-    using Matrix = Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>;
+    using Matrix = Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
```

Comment:
In case you forgot it somewhere it would crash right?

I.e. it would not silently make copies?

Reply:
I am not sure if it would crash, Eigen can process combination of row and col_major matrices and does that silently (as far as i know). I think it depends on the operation if Eigen makes an internal copy.

It is important however to get this right when interfacing ublas matrices, if the matrix is square and symmetric it would eventually even give the right results, it would give wrong results if not. I think we have to rely on our tests for that.